### PR TITLE
refactor: jest matchers and browser test types

### LINKED ISSUE #67: Warnings in linter output
When running `npm run lint` (or with `:check`) there are warnings in the output:

```
  35:44  warning  Unexpected 'todo' comment: 'TODO: write own matcher to check swarm...'  no-warning-comments
  40:38  warning  Unexpected 'todo' comment: 'TODO: write own matcher to handle Bee...'   no-warning-comments
```

It would be good if there were no warnings when we run the linter.

### LINKED ISSUE #117: Browser test's types have broken
During the time, the custom types that were made for browser tests (in types/global.d.ts) does not load and also from PR merge #115, it became outdated.

Neccesary to define new types for browser tests according to the current state.

### PR DESCRIPTION
**Pull request by nugaon**
* Fixing type issues in browser tests
* amend README about tests
* make examples for creating own Jest Matchers for tests, in order to better test readability and code quality  

Fixes #117, resolves #67

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -2,6 +2,41 @@ import { Readable } from 'stream'
...
+ */
+export function commonMatchers(): void {
+  expect.extend({
+    toBeHashReference(received: string) {
+      const result = {
+        pass: false,
+        message: () => 'The given object was not a Swarm hash reference',
+      }
+
+      if (typeof received === 'string' && received.length === 64) {
```
Comment (vojtechsimetka) [inline test/utils.ts:16]:
```suggestion
      if (typeof received === 'string' && /^[0-9a-fA-F]{64}$/.test(received)) {
```

File: test/utils.ts [hunk 2]
```diff
@@ -2,6 +2,41 @@ import { Readable } from 'stream'
...
 
+/**
+ * Load common own Jest Matchers which can be used to check particular return values.
+ */
+export function commonMatchers(): void {
+  expect.extend({
+    toBeHashReference(received: string) {
+      const result = {
+        pass: false,
+        message: () => 'The given object was not a Swarm hash reference',
```
Comment (vojtechsimetka) [inline test/utils.ts:13]:
```suggestion
        message: () => 'Given input is not a Swarm hash reference',
```

File: test/utils.ts [hunk 3]
```diff
@@ -2,6 +2,41 @@ import { Readable } from 'stream'
...
+      }
+
+      if (typeof received === 'string' && received.length === 64) {
+        result.pass = true
+        result.message = () => 'The given string has correct hash length'
```
Comment (vojtechsimetka) [inline test/utils.ts:18]:
```suggestion
        result.message = () => 'Given string is semantically valid Swarm hash reference'
```

File: tsconfig.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
     "resolveJsonModule": true,
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
-    "typeRoots": ["./src/types"]
+    "typeRoots": ["./src/types", "./test/types"]
```
Comment (vojtechsimetka) [inline tsconfig.json:14]: Why is this needed? These are only test types, right?

File: README.md [hunk 5]
```diff
@@ -114,8 +114,7 @@ There are also browser tests by Puppeteer, which also provide integrity testing.
 npm run test:browser
 ```
 The test HTML file which Puppeteer uses is the [test/testpage/testpage.html](test/testpage/testpage.html).
-In order to open this file and manually test BeeJS via developer console, it's necessary to build the test JS file that it imports with `npm run compile:window`.
-This test JS file is identical with the Bee library itself, except it binds BeeJS functionalities to the browser's `window` object.
+In order to open this file and manually test BeeJS via developer console, it's necessary to build the test JS file that it imports with `npm run compile:browser` (also running `npm run test:browser` creates the necessary library files).
```
Comment (vojtechsimetka) [inline README.md:117]:
```suggestion
To open and manually test BeeJS with developer console, it is necessary to build the library first with `npm run compile:browser` (running the browser tests `npm run test:browser` also builds the library).
```

**Review by vojtechsimetka — APPROVED** [review]

File: tsconfig.json (hunk 4 above)
Comment (nugaon) [reply tsconfig.json:14]: because without it the IDE throws error on opening test files
